ext/nnstreamer/tensor_decoder/tensordec-protobuf.cc

```diff
@@ -0,0 +1,177 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer / NNStreamer tensor_decoder subplugin, "protobuf"
+ * Copyright (C) 2020 Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ */
+/**
+ * @file        tensordec-protobuf.cc
+ * @date        25 Mar 2020
+ * @brief       NNStreamer tensor-decoder subplugin, "protobuf",
+ *              which converts tensor or tensors to Protocol Buffers.
+ * @see         https://github.com/nnstreamer/nnstreamer
+ * @author      Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+#include <glib.h>
+#include <gst/gstinfo.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer_log.h>
+
+#include "nnstreamer.pb.h"    /* Generated by `protoc` */
+
+void init_pb (void) __attribute__ ((constructor));
+void fini_pb (void) __attribute__ ((destructor));
+
+#define DECODER_PB_BIN_CAPS_STR "application/octet-stream"
```

Comment:
You will need to change when you create a converter subplug-in.

Reply:
Right! I added a caps for protobuf-converter/decoder in tensor_typedef.h after you.